File: message/sample/message_sample/src/main/java/com/sphere/sample/message/MyFirebaseMessagingService.java
package com.sphere.sample.message;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.sphere.message.SpherePushMessage;

public class MyFirebaseMessagingService extends FirebaseMessagingService {

    @Override
    public void onNewToken(@NonNull String token) {
        // Sphere SDK 푸시 토큰 설정
        SpherePushMessage.setFcmToken(token);
    }

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        if (SpherePushMessage.isSpherePushMessage(remoteMessage.getData())) {
            // Sphere 푸시 메시지 데이터 처리: 앱이 실행 중인 경우 알림창에 메시지 표시
            SpherePushMessage.handleMessageReceived(remoteMessage.getData());

        } else {
            // Sphere 푸시 메시지가 아닌 경우 처리
        }
    }
}

File: message/sample/message_sample_kt/src/main/java/com/sphere/sample/message/kt/MyFirebaseMessagingService.kt
package com.sphere.sample.message.kt

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.sphere.message.SpherePushMessage

class MyFirebaseMessagingService : FirebaseMessagingService() {

    override fun onNewToken(token: String) {
        // Sphere SDK 푸시 토큰 설정
        SpherePushMessage.setFcmToken(token)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        if (SpherePushMessage.isSpherePushMessage(remoteMessage.data)) {
            // Sphere 푸시 메시지 데이터 처리: 앱이 실행 중인 경우 알림창에 메시지 표시
            SpherePushMessage.handleMessageReceived(remoteMessage.data)

        } else {
            // Sphere 푸시 메시지가 아닌 경우 처리
        }
    }
}
